src/redisraft.c

```diff
@@ -575,14 +586,18 @@ static void handleRedisCommand(RedisRaftCtx *rr,
     }
 
     /* Confirm that we're the leader and handle redirect or proxying if not. */
-    Node **ret = rr->config->follower_proxy ? &leader_proxy : NULL;
+    Node **ret = rr->config->follower_proxy || cmds->asking ? &leader : NULL;
     if (checkLeader(rr, ctx, ret) == RR_ERROR) {
         return;
     }
 
     /* Proxy */
-    if (leader_proxy) {
-        if (ProxyCommand(rr, ctx, cmds, leader_proxy) != RR_OK) {
+    if (leader) {
+        if (cmds->asking) {
+            if (computeHashSlot(rr, ctx, cmds) == RR_OK){
```

Comment:
What do we do if `computeHashSlot()` returns an error? We have to reply.

Reply:
`computeHashSlot` was changed to support replying in error, but I think we should avoid that.

Reply:
Ok, I see this branch lags behind the master. @sjpotter has already changed this behavior: https://github.com/RedisLabs/redisraft/pull/331

Reply:
Well, I'm not sure, because it's part of the changes in the PR... anyway, we all agree that `computeHashSlot` should not reply in case of error... 🙏